src/app/login/login/login.component.ts

```diff
@@ -23,8 +23,20 @@ export class LoginComponent implements OnInit {
     this.authService.login(this.userName, this.password)
       .subscribe((result) => {
         this.errorMessage = null;
-        // TODO: redirect user to role-specific route
-        this.router.navigate(['/shell/admin-panel/']);
+        let route: string;
```

Comment:
move repeated code to the function in auth service and call in needed places

Reply:
Function defaultRoute was created and code was moved to the fucntion